### PR TITLE
Switch from fail2ban to sshguard

### LINKED ISSUE #2221: Implement enabling PerSourcePenalties with SSHguard
The purpose of this issue is to follow up on https://github.com/gardenlinux/security/issues/26 with configuring Garden Linux going forward to use PerSourcePenalties with SSHguard.

@gehoern will go for it.

CC @Akendo

### PR DESCRIPTION
**Pull request by fwilhe**
**What this PR does / why we need it**:
The decision was made to switch from fail2ban to sshguard.
This PR replaces the packages and removes fail2ban config.

Also the Garden Linux apt repo from tomorrow on contains  openssh 9.8p1-2 which includes the PerSourcePenalties.
This is enabled by default, custom configuration can be made.

**Which issue(s) this PR fixes**:
Fixes #2221

**Reviewer notes:**

I'm not sure if we need special configuration for sshguard. We had quite some config for fail2ban which is deleted in this PR.

I did not find any specific tests for fail2ban, not sure if we need tests for sshguard.

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-today
```
Comment (fwilhe) [inline VERSION:1]: needs to be reverted once sshguard is in the today dist (tomorrow)

File: features/ssh/exec.config [hunk 2]
```diff
@@ -1,5 +1,34 @@
 #!/usr/bin/env bash
 set -Eeuo pipefail
+set -x
```
Comment (Vincinator) [inline features/ssh/exec.config:3]: do we need this?

Comment (Vincinator) [inline features/ssh/exec.config:3]:
```suggestion
```

Comment (fwilhe) [reply features/ssh/exec.config]: thanks, forgot to remove after debugging

**Review by 5kt — APPROVED** [review]

**Pull request merged**